tests/plenary/state/state_spec.lua

```diff
@@ -0,0 +1,144 @@
+local utils = require('orgmode.utils')
+local state = nil
+local cache_path = vim.fs.normalize(vim.fn.stdpath('cache') .. '/org-cache.json', { expand_env = false })
+
+describe('State', function()
+  before_each(function()
+    -- Ensure the cache file is removed before each run
+    state = require('orgmode.state.state')
+    vim.wait(50, function()
+      return state._ctx.saved
+    end, 10)
+    state:wipe()
+    vim.fn.delete(cache_path, 'rf')
+  end)
+
+  after_each(function()
+    state._ctx.saved = false
+    state._ctx.loaded = false
+    state = nil
+    -- vim.fn.delete(cache_path, 'rf')
+  end)
+
+  it("should create a state file if it doesn't exist", function()
+    -- Ensure the file doesn't exist
+    vim.fn.delete(cache_path, 'rf')
+    local stat = vim.loop.fs_stat(cache_path)
+    if stat then
+      error('Cache file existed before it should! Ensure it is deleted before each test run!')
+    end
+
+    -- This creates the cache file on new instances of `State`
+    state:load()
```

Comment:
Use `load_sync` here and remove the `vim.wait` below.
```suggestion
    state:load_sync()
```

Reply:
So this test is actually not checking the precise usage of `State:load`. It's validating that the cache file is created by `State:save` via a `State:load` call. As such `State:load_sync` doesn't actually have any value here.

This test is kinda peeking under the hood a bit to validate some "magic" functionality of `State:load`. It's important that the `vim.wait` stays. Notice that it's checking the status of `state._ctx.saved`, not `state._ctx.loaded`.

Reply:
Ok, that makes sense. Thanks for clarifying!